model/src/main/java/org/projectnessie/model/Branch.java

```diff
@@ -33,8 +35,15 @@
 @JsonTypeName("BRANCH")
 public interface Branch extends Reference {
 
+  @Override
+  @NotBlank
+  @Pattern(regexp = Validation.REF_NAME_REGEX, message = Validation.REF_NAME_MESSAGE)
+  @Value.Parameter(order = 1)
+  String getName();
+
   @Override
   @Nullable
+  @Value.Parameter(order = 2)
   String getHash();
```

Comment:
IMHO, using `of` method parameters of the same type (`String` in this case) is prone to mistakes... Are the runtime savings great enough to risk that?

To clarify: I'm ok with `of` methods where parameters are of different types.

Reply:
In this particular case, this is a moot point since we already have an `of(String, String)` method ;)